### PR TITLE
forklift: fix race between prime pvc and pod removal

### PR DESCRIPTION
**Pull request by bennyz**
Because there is time until deletionTimestamp is set on the PVC prime, and since the populator pod is deleted first we might reconcile again when the PVC is not yet marked for deletion, causing the populator pod to be recreated.

This change reorders the operation, to first wait until the PVC is marked for deletion, and then delete the populator pod.

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
Fix race in the forklift populator that might prevent cleanup
```

### CODE REVIEW

File: pkg/controller/populators/forklift-populator.go [hunk 1]
```diff
@@ -212,6 +207,10 @@ func (r *ForkliftPopulatorReconciler) reconcile(ctx context.Context, req reconci
 		res, err = r.reconcileCleanup(pvcPrime)
 	}
 
+	if pvcPrime.DeletionTimestamp != nil {
+		res, err = r.deletePopulatorPod(fmt.Sprintf("%s-%s", populatorPodPrefix, pvc.UID), pvc, pvcPrime)
```
Comment (mhenriks) [inline pkg/controller/populators/forklift-populator.go:211]: in what cases would pvc prime be deleted before the pod?

Comment (bennyz) [reply pkg/controller/populators/forklift-populator.go:211]: `res, err = r.reconcileCleanup(pvcPrime)` in L#207 marks it for deletion

Comment (mhenriks) [reply pkg/controller/populators/forklift-populator.go:211]: Why not have `reconcileCleanup` delete the pod as well then?

**Comment by bennyz** [comment]
> Is this change covered by a unit test? I ask because coverage for forklift-populator.go is very low at 24.9%

I'll add

**Comment by mhenriks** [comment]
/lgtm
/approve

@bennyz nice bump in test coverage, more would be welcome in subsequent prs!

**Pull request merged**